task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -862,6 +862,7 @@ def _send_heartbeat_if_needed(self):
                     ti_id=self.id,
                 )
                 self.kill(signal.SIGTERM, force=True)
+                raise
```

Comment:
Raisng here means task failure callbacks etc don't get a chance to run doesn't it? Or we won't upload task logs etc?

Reply:
removed